src/full/Agda/TypeChecking/Records.hs

```diff
@@ -345,6 +345,12 @@ isEtaCon c = getConstInfo' c >>= \case
     Constructor {conData = r} -> isEtaRecord r
     _ -> return False
 
+-- | Going under one of these does not count as a decrease in size for the termination checker.
+isEtaOrCoinductiveRecordConstructor :: HasConstInfo m => QName -> m Bool
+isEtaOrCoinductiveRecordConstructor c =
+  caseMaybeM (isRecordConstructor c) (return False) $ \ (_, def) ->
+    return $ (Just Inductive, NoEta) /= (recInduction def, recEtaEquality def)
```

Comment:
What about `(Nothing, Eta)`?  This would be a (by default) inductive record constructor with eta.

Reply:
It returns `True` for `(Nothing, Eta)` as you would expect.